### DISCUSSION TITLE
Sometimes hide the edit button on a folder

### DISCUSSION THREAD

**Discussion by woodie** [gollum/gollum#1827] [Ideas]
I'd like to add `.authors` files in directories to then hide the `Edit` button for users not in there. Does a feature like that already exist? Maybe we can override a method and include...
```rb
@allow_editing = user_can_edit  # OR
settings.wiki_options[:allow_editing] = user_can_edit
```

**Comment by srbaker** [comment]
I suspect the best solution here is to integrate Pundit via OmniGollum?  I'd be willing to collaborate on this if you'd like.  I think you can find my email via GitHub.

**Comment by woodie** [comment]
I see how they alter the wiki_options [here](https://stackoverflow.com/questions/40690778/how-to-configure-gollum-wiki-to-allow-basic-authentication-and-to-use-the-userna), but I'm using OAuth and my [config.rb](https://github.com/woodie/gollum-wiki-oauth/blob/main/config.rb) may not be accessing the user/email early enough. You can see my setup [here](https://github.com/woodie/gollum-wiki-oauth/).

**Comment by dometto** [comment]
Converted this to a Discussion. Please feel free to post here if you got this to work, or are still working on it.
